unittests/test_dependencies.py

```diff
@@ -112,140 +115,360 @@ def test_eq_hash(loader, exec_ctx):
+                    Node('Test1_custom', 'sys0:p0', 'e0'),
+                    Node('Test0', 'sys0:p1', 'e1'))
 
     # Check in-degree of Test0
 
-    # 2 from Test1_fully,
-    # 1 from Test1_by_env,
-    # 1 from Test1_exact,
+    # 4 from Test1_always,
+    # 2 from Test1_part_equal,
+    # 1 from Test1_part_env_equal,
+    # 2 from Test1_any,
+    # 2 from Test1_all,
+    # 0 from Test1_custom,
     # 1 from Test1_default
-    assert in_degree(deps, Node('Test0', 'sys0:p0', 'e0')) == 5
-    assert in_degree(deps, Node('Test0', 'sys0:p1', 'e0')) == 5
-
-    # 2 from Test1_fully,
-    # 1 from Test1_by_env,
-    # 2 from Test1_exact,
+    assert in_degree(deps, Node('Test0', 'sys0:p0', 'e0')) == 12
+
+    # 4 from Test1_always,
+    # 2 from Test1_part_equal,
+    # 1 from Test1_part_env_equal,
+    # 2 from Test1_any,
+    # 0 from Test1_all,
+    # 0 from Test1_custom,
     # 1 from Test1_default
-    assert in_degree(deps, Node('Test0', 'sys0:p0', 'e1')) == 6
-    assert in_degree(deps, Node('Test0', 'sys0:p1', 'e1')) == 6
+    assert in_degree(deps, Node('Test0', 'sys0:p1', 'e0')) == 10
+
+    # 4 from Test1_always,
+    # 2 from Test1_part_equal,
+    # 1 from Test1_part_env_equal,
+    # 4 from Test1_any,
+    # 0 from Test1_all,
+    # 0 from Test1_custom,
+    # 1 from Test1_default
+    assert in_degree(deps, Node('Test0', 'sys0:p0', 'e1')) == 12
+
+    # 4 from Test1_always,
+    # 2 from Test1_part_equal,
+    # 1 from Test1_part_env_equal,
+    # 4 from Test1_any,
+    # 0 from Test1_all,
+    # 1 from Test1_custom,
+    # 1 from Test1_default
+    assert in_degree(deps, Node('Test0', 'sys0:p1', 'e1')) == 13
 
     # Pick a check to test getdep()
-    check_e0 = find_case('Test1_exact', 'e0', cases).check
-    check_e1 = find_case('Test1_exact', 'e1', cases).check
+    check_e0 = find_case('Test1_part_equal', 'e0', 'p0', cases).check
+    check_e1 = find_case('Test1_part_equal', 'e1', 'p0', cases).check
 
     with pytest.raises(DependencyError):
-        check_e0.getdep('Test0')
+        check_e0.getdep('Test0', 'p0')
 
     # Set the current environment
     check_e0._current_environ = Environment('e0')
     check_e1._current_environ = Environment('e1')
 
-    assert check_e0.getdep('Test0', 'e0').name == 'Test0'
-    assert check_e0.getdep('Test0', 'e1').name == 'Test0'
-    assert check_e1.getdep('Test0', 'e1').name == 'Test0'
+    assert check_e0.getdep('Test0', 'e0', 'p0').name == 'Test0'
+    assert check_e0.getdep('Test0', 'e1', 'p0').name == 'Test0'
+    assert check_e1.getdep('Test0', 'e1', 'p0').name == 'Test0'
     with pytest.raises(DependencyError):
-        check_e0.getdep('TestX', 'e0')
+        check_e0.getdep('TestX_deprecated', 'e0', 'p0')
 
     with pytest.raises(DependencyError):
-        check_e0.getdep('Test0', 'eX')
+        check_e0.getdep('Test0', 'eX', 'p0')
 
     with pytest.raises(DependencyError):
-        check_e1.getdep('Test0', 'e0')
-
-
-def test_build_deps_unknown_test(loader, exec_ctx):
```

Comment:
We still need this test. We still expect to get a dependency error.

Reply:
What I mean here is that we need to check that if we pass the `how` argument the old way, (a) we get deprecation warning and (b) it creates the correct dependencies. You just need to check in a similar way as in your other unit test, except that you only try the three different parameters for `how`.